proto-lens/src/Data/ProtoLens/Encoding.hs

```diff
@@ -89,15 +89,15 @@ parseAndAddField :: msg
 parseAndAddField
     msg
     (FieldDescriptor name typeDescriptor accessor)
-    (TaggedValue tag (WireValue wt val))
+    (TaggedValue tag (WireValue (wt :: WireType w) val))
     = case fieldWireType typeDescriptor of
-        FieldWireType fieldWt _ get -> let
+        (FieldWireType fieldWt _ get :: FieldWireType a) -> let
           getSimpleVal = do
               Equal <- equalWireTypes name fieldWt wt
               get val
           -- Get a block of packed values, reversed.
+          getPackedVals :: w ~ B.ByteString => Either String [a]
```

Comment:
Would this be simpler as "getPackedVals :: B.ByteString -> Either String [a]" and then the type equality can be implicit at the call site (in the RepeatedField case below)?